transfer-archive/app/connectors/sas_generator/sas_generator_client.py

```diff
@@ -0,0 +1,39 @@
+import logging
+from typing import Optional
+
+from httpx import AsyncClient, HTTPError
+
+from app.connectors.sas_generator.exceptions import StorageSASGeneratorRequestFailed
+from app.connectors.sas_generator.models import SASTokenRequest, SASResponse
+
+logger = logging.getLogger(__name__)
+
+
+class SASGeneratorClient:
+    def __init__(self, sas_generator_base_url: str):
+        self.url = f"{sas_generator_base_url}/generate_sas"
+
+    async def request_sas(self, container_id: str, duration_hours: int = 24) -> Optional[SASResponse]:
+        async with AsyncClient() as client:
+            request = SASTokenRequest(container=container_id, duration_hours=duration_hours)
+
+            try:
+                logger.info(f"Requesting SAS Token from {self.url} with payload: {request.dict()}")
+                resp = await client.post(self.url, json=request.dict())
+            except HTTPError as err:
```

Comment:
Bytt ut `HTTPError` med `RequestError` og `HTTPStatusError`, siden `HTTPError` ikke har attribute `request`, så får man en type error.

```suggestion
from httpx import AsyncClient, RequestError, HTTPStatusError

from app.connectors.sas_generator.exceptions import StorageSASGeneratorRequestFailed
from app.connectors.sas_generator.models import SASTokenRequest, SASResponse

logger = logging.getLogger(__name__)


class SASGeneratorClient:
    def __init__(self, sas_generator_base_url: str):
        self.url = f"{sas_generator_base_url}/generate_sas"

    async def request_sas(self, container_id: str, duration_hours: int = 24) -> Optional[SASResponse]:
        async with AsyncClient() as client:
            request = SASTokenRequest(container=container_id, duration_hours=duration_hours)

            try:
                logger.info(f"Requesting SAS Token from {self.url} with payload: {request.dict()}")
                resp = await client.post(self.url, json=request.dict())
            except (RequestError, HTTPStatusError) as err:
```